sql/colexec: TestNotExprProjOp failed

sql/colexec.TestNotExprProjOp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15243457?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15243457?buildTab=artifacts#/) on release-23.2.5-rc @ [57ab10786d6ed277959c8990e2823772893ac873](https://github.com/cockroachdb/cockroach/commits/57ab10786d6ed277959c8990e2823772893ac873):


```
        github.com/cockroachdb/cockroach/pkg/sql/colexec.TestNotExprProjOp(0xc00090d6c0)
        	github.com/cockroachdb/cockroach/pkg/sql/colexec/not_expr_ops_test.go:91 +0x19c4
        testing.tRunner(0xc00090d6c0, 0x355ce78)
        	GOROOT/src/testing/testing.go:1595 +0xff
        created by testing.(*T).Run in goroutine 1
        	GOROOT/src/testing/testing.go:1648 +0x3ad
        goroutine 1 [chan receive]:
        testing.(*T).Run(0xc0004c2d00, {0x337871d?, 0x607f5c?}, 0x355ce78)
        	GOROOT/src/testing/testing.go:1649 +0x3c8
        testing.runTests.func1(0x6101060?)
        	GOROOT/src/testing/testing.go:2054 +0x3e
        testing.tRunner(0xc0004c2d00, 0xc000559318)
        	GOROOT/src/testing/testing.go:1595 +0xff
        testing.runTests(0xc000656320?, {0x5e12e20, 0x3f, 0x3f}, {0x4d?, 0x80?, 0x60fb6e0?})
        	GOROOT/src/testing/testing.go:2052 +0x445
        testing.(*M).Run(0xc000656320)
        	GOROOT/src/testing/testing.go:1925 +0x636
        github.com/cockroachdb/cockroach/pkg/sql/colexec.TestMain.func1(0x33a4568?)
        	github.com/cockroachdb/cockroach/pkg/sql/colexec/main_test.go:81 +0x57b
        github.com/cockroachdb/cockroach/pkg/sql/colexec.TestMain(0x33a4568?)
        	github.com/cockroachdb/cockroach/pkg/sql/colexec/main_test.go:82 +0x25
        main.main()
        	main/bazel-out/k8-fastbuild/bin/pkg/sql/colexec/colexec_test_/testmain.go:310 +0x512
        goroutine 50 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:172 +0xfd
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:164 +0x1a
        goroutine 10 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:117 +0x4e
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:83 +0x1a
        goroutine 11 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:141 +0x89
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:84 +0x26
        goroutine 37 [syscall]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x29
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x13
        created by os/signal.Notify.func1.1 in goroutine 11
        	GOROOT/src/os/signal/signal.go:151 +0x1f: Leaked goroutine: goroutine 50 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:172 +0xfd
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:164 +0x1a
--- FAIL: TestNotExprProjOp (5.04s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestNotExprProjOp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-38692

--- a/pkg/util/leaktest/leaktest.go
+++ b/pkg/util/leaktest/leaktest.go
@@ -70,6 +70,9 @@ func interestingGoroutines() map[int64]string {
 			// duration of the process.
 			strings.Contains(stack, "log.flushDaemon") ||
 			strings.Contains(stack, "log.signalFlusher") ||
+			// Ignore the goschedstats goroutine that lives for the duration of
+			// the process.
+			strings.Contains(stack, "util/goschedstats.init") ||
 			// Below are the stacks ignored by the upstream leaktest code.
 			strings.Contains(stack, "testing.Main(") ||
 			strings.Contains(stack, "testing.tRunner(") ||
@@ -172,7 +175,7 @@ func AfterTest(t T) func() {
 	}
 }
 
-// diffGoroutines compares the current goroutines with the base snapshort and
+// diffGoroutines compares the current goroutines with the base snapshot and
 // returns an error if they differ.
 func diffGoroutines(base map[int64]string) error {
 	var leaked []string
@@ -188,7 +191,7 @@ func diffGoroutines(base map[int64]string) error {
 	sort.Strings(leaked)
 	var b strings.Builder
 	for _, g := range leaked {
-		b.WriteString(fmt.Sprintf("Leaked goroutine: %v\n\n", g))
+		b.WriteString(fmt.Sprintf("\nLeaked goroutine: %v\n", g))
 	}
 	return errors.Newf("%s", b.String())
 }

